qt/python/mantidqt/plotting/functions.py

```diff
@@ -162,6 +162,11 @@ def pcolormesh(workspaces, fig=None):
         if subplot_idx < workspaces_len:
             ws = workspaces[subplot_idx]
             pcm = pcolormesh_on_axis(ax, ws)
+            if ax.images:  # Colour bar limits are wrong if workspace is ragged. Set them manually.
+                colorbar_min = np.nanmin(ax.images[-1].get_array())
+                colorbar_max = np.nanmax(ax.images[-1].get_array())
+                for image in ax.images:
+                    image.set_clim(colorbar_min, colorbar_max)
```

Comment:
The `pcm` variable above refers to the image that's just been created, so you could just do `pcm.get_array()` rather than `ax.images[-1].get_array()`.

Also is there a reason you're setting the limits for every image in the axes as opposed to solely setting them for `pcm`?

Reply:
> Also is there a reason you're setting the limits for every image in the axes as opposed to solely setting them for pcm?

It was for if there were tiled plots, but if that's not something we really support I can change it.

Reply:
Tiled colorfill plots are *technically* possible but they don't work very well, the colorbar is only linked to to the last plot instead of all of them (or one colorbar for each plot).

In any case, this code executes for each plot so wouldn't it make more sense to only set the limits for the image that's just been created instead of all of them? Then every image in the tiled plot would have appropriate limits for their data rather than them all having the same limits whether they're suitable or not.

Reply:
Ah, I think I misunderstood how the code worked here. You're right, I'll make those changes.